Zotero 7

thebluepotato opened the pull request:
This PR brings support for DOI Manager to Zotero 7, related to #41.

This should be considered very much WIP.

Current features:
- [x] Item menu functions (lookup short/long DOIs, cleanup and verify)
- [x] Tools menu functions (change setting for new items)
- [x] Preference window
- [x] Check update logic (JSON vs RDF)
- [x] Check support for Zotero versions below 5?

~Lots of debugging required, but seems to work both in Zotero 6 and 7.~

Note that the core logic of fetching/checking DOIs is unchanged. The aim here is to get a version working with Zotero 7, with the same underlying logic.

thebluepotato committed:
Getting a pre-alpha semi-working version out

This commit brings support for DOI Manager to Zotero 7, albeit in a very pre-alpha way yet.

bwiernik commented:
Awesome thank you! I will take a look. Please feel free to keep adding to this PR

thebluepotato committed:
Adding preferences and cleanup

thebluepotato commented:
Preferences window is up and running, now part of the Zotero preferences instead of its own window. Now using Fluent for that preference pane. Since the tools and item menus are now generated via code instead of overlays (for Zotero 6 & 7), certain obsolete language files were deleted.

thebluepotato committed:
Additional cleanup

thebluepotato committed:
Updated updates mechanism

Updated version will already point to the the new JSON updates format for Zotero 6 so the add-on will work seamlessly when users update to Zotero 7

thebluepotato commented:
I guess the updating mechanism should work? Z6 will install based on `install.rdf` which now points to `updates.json` for updates, just like Z7's `manifest.json`. Add-on is no longer compatible with Z5 because I'm pretty sure Zotero's new system is not meant to be compatible with 3 major versions simultaneously...

LionSR commented:
curious to know if this is ready to go?

thebluepotato commented:
> curious to know if this is ready to go?

The code has been ready for a while now, and the update mechanism should work (no real way to test it). I'm just waiting for @bwiernik to merge the PR

Mengchuxia commented:
so cooooool

citadel-lewis commented:
@thebluepotato is it possible just to create a fork of this project using your code? It's been 2 months waiting for this to be merged.

thebluepotato commented:
> @thebluepotato is it possible just to create a fork of this project using your code? It's been 2 months waiting for this to be merged.

You can use the fork this PR is based on. I haven't tested the "script" that packages the code into XPI but it should be trivial (zip the files and rename the extension).

citadel-lewis commented:
> > @thebluepotato is it possible just to create a fork of this project using your code? It's been 2 months waiting for this to be merged.
> 
> You can use the fork this PR is based on. I haven't tested the "script" that packages the code into XPI but it should be trivial (zip the files and rename the extension).

Alright, I'll try this tomorrow. Thanks!

pkej commented:
Change extension from ".zip" to ".xpi". [zotero-doi-manager-1.5.0.zip](https://github.com/user-attachments/files/16201874/zotero-doi-manager-1.5.0.zip)

citadel-lewis commented:
> Change extension from ".zip" to ".xpi". [zotero-doi-manager-1.5.0.zip](https://github.com/user-attachments/files/16201874/zotero-doi-manager-1.5.0.zip)

That installs okay but when trying to select DOI Manager tab in the settings it doesn't load the tab to see the settings ... not sure if it's working or not

thebluepotato committed:
Some cleanup

thebluepotato committed:
Moved options pane and removed isCollection test

thebluepotato commented:
I just pushed some changes that should fix the missing prefpane. I had moved it to another folder without updating the link in code. For some reason, one of the checks made when updating DOIs (namely that the current item is not a collection) threw an error so I also removed it since if the item is a "regular item" it cannot be a collection (from the sidebar) at the same time. Please feel free to test and report back!

thebluepotato committed:
Update (untested) build script

mjthoraval commented:
This is working great for me.
Thank you very much!
The emoji tags are now displayed in the Items Tree in the latest beta beta version: https://github.com/zotero/zotero/commit/26f7c707bae46534d19be08907f54b2f3c798ed8

mjthoraval commented:
The plugin does not seem to work on newly added items. For example, importing this article does not find a DOI, but the plugin does not add the "No DOI found" tag:
https://www.jkps.or.kr/journal/view.html?spage=1335&volume=49&number=4

Here are the options I have set:

![image](https://github.com/user-attachments/assets/cd852760-4ba8-40d8-9896-1db34c6f9dcd)

thebluepotato committed:
Fix autoretrieve logic + cleanup

thebluepotato commented on chrome/content/scripts/zoteroshortdoi.js:
```diff
@@ -1,493 +0,0 @@
-            case "long":
-              Zotero.ShortDOI.updateItems(Zotero.Items.get(ids), "long");
-              break;
-            case "verify":
```
This check was already buggy, since the autoretrieve setting would be set to `check` and not `verify`.

thebluepotato commented:
@mjthoraval Should be fixed now. This was actually buggy in the Zotero 6 version, I guess.

mjthoraval commented:
Problem fixed, thank you.

SR-- commented:
Is the binary above the one with the latest fix, please?

thebluepotato commented:
> Is the binary above the one with the latest fix, please?

I guess you're referring to [this comment](https://github.com/bwiernik/zotero-shortdoi/pull/42#issuecomment-2226855938). There you go for the latest version: 
[zotero-doi-manager-1.5.0.zip](https://github.com/user-attachments/files/16558766/zotero-doi-manager-1.5.0.zip). You should download the zip file and then rename it to xpi.

thebluepotato commented:
@bwiernik Any ETA on merging this PR? As noted above, people have begun using it. Dunno whether the update mechanism works, but only one way to find out.

SR-- commented:
> > Is the binary above the one with the latest fix, please?
> 
> I guess you're referring to [this comment](https://github.com/bwiernik/zotero-shortdoi/pull/42#issuecomment-2226855938). There you go for the latest version: [zotero-doi-manager-1.5.0.zip](https://github.com/user-attachments/files/16558766/zotero-doi-manager-1.5.0.zip). You should download the zip file and then rename it to xpi.

Thanks. Does not work for me. Installs but fails to update a long DOI into a short one.

Shows this window but never completes.

![image](https://github.com/user-attachments/assets/239987ed-ffa2-4458-a72e-8516fb5c3879)

thebluepotato commented:
If you want me to debug it I'll need the logs and the item you're trying to update

SR-- commented:
Not item-specific. None works. Is this relevant?
![image](https://github.com/user-attachments/assets/7c45fe11-5fb5-4d50-ba95-abd19e7e0ce6)

thebluepotato commented:
> Not item-specific. None works. Is this relevant? ![image](https://private-user-images.githubusercontent.com/1008059/356622465-7c45fe11-5fb5-4d50-ba95-abd19e7e0ce6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMyMTQ0OTcsIm5iZiI6MTcyMzIxNDE5NywicGF0aCI6Ii8xMDA4MDU5LzM1NjYyMjQ2NS03YzQ1ZmUxMS01ZmI1LTRkNTAtYmE5NS1hYmQxOWU3ZTBjZTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MDlUMTQzNjM3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWI2MTUxMDRlNTMwMWQyYmJjNGM3YTkyYjBiZDE2MTlhNDk5NGRlODVlMGYxZDZkMWQwYTZmYTFjYzIyM2RmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yozLTV1kMN_RJVrAXXbVl5_VhViEO63agZc5uUpQXgQ)

Are you sure you're using the ZIP file from my comment, and not @pkej's? The `isCollection` lines have all been removed. I've checked with my test entries and have been able to switch back and forth between long and short DOIs.

thebluepotato committed:
Update .gitignore

bwiernik commented:
Thanks @thebluepotato

SR-- commented:
@thebluepotato

I have installed the released version now, but the result is the same. The process does not complete, and I am seeing the same error.
---
I am so confused because the file I download has no such lines but the one in Zotero profile has this:

![image](https://github.com/user-attachments/assets/2b5c7f6e-d206-449c-932e-2d9404d1b6bd)

Actually, I see that the released ZIP does not have such lines but XPI does...

thebluepotato commented:
@SR-- that's odd indeed 😅
The new release script that was merged will definitely help. Trying to automate this even more with GitHub Actions but I'm really unfamiliar with them. Anyway, you can actually take the "released ZIP" and rename it to XPI and you're good to go. There's a few extra files in there but you should be fine.

@bwiernik I think you made the XPI on the basis of a stale local copy of the repo. Using the `release.sh` will guarantee that you've pulled the latest commit for now, but I'll still try to suggest a GitHub action so in the future you just have to click release or something.

SR-- commented:
@thebluepotato Thank for looking into this. I have tried renaming the ZIP already actually, but Zotero rejected it.

mjthoraval commented:
I am using the latest files in this fork, identified as version 1.5.1.
The tag "⛔ No DOI found" is correctly applied when importing the article from the page I had reported above:
https://www.jkps.or.kr/journal/view.html?spage=1335&volume=49&number=4
But it does not work for this page, which also does not have any DOI:
https://caves.org/journal-of-cave-and-karst-studies/jcks-articles/variable-calcite-deposition-rates-as-proxy-for-paleo-precipitation-determination-as-derived-from-speleothems-in-central-florida-u-s-a/

After adding the item in my library, the manual check "Verify and clean DOIs" also does not work.
The Debug Output shows:
`(3)(+0000002): Zotero DOI Manager: CrossRef lookup: Unknown status code: malformed`

Tested on:
Zotero 7.0.0-beta.117+794dc6bf0 (64-bit)
Windows 10

thebluepotato commented:
> I am using the latest files in this fork, identified as version 1.5.1. The tag "⛔ No DOI found" is correctly applied when importing the article from the page I had reported above: https://www.jkps.or.kr/journal/view.html?spage=1335&volume=49&number=4 But it does not work for this page, which also does not have any DOI: https://caves.org/journal-of-cave-and-karst-studies/jcks-articles/variable-calcite-deposition-rates-as-proxy-for-paleo-precipitation-determination-as-derived-from-speleothems-in-central-florida-u-s-a/
> 
> After adding the item in my library, the manual check "Verify and clean DOIs" also does not work. The Debug Output shows: `(3)(+0000002): Zotero DOI Manager: CrossRef lookup: Unknown status code: malformed`
> 
> Tested on: Zotero 7.0.0-beta.117+794dc6bf0 (64-bit) Windows 10

The CrossRef lookup needs an author or a page to check DOIs. The translator imports the article you mentioned into the library without either, so the plugin fails instead of tagging the item with "No DOI".

SR-- commented:
@thebluepotato 

> Anyway, you can actually take the "released ZIP" and rename it to XPI and you're good to go. There's a few extra files in there but you should be fine.

Tried again, does not work. What am I missing?

thebluepotato commented:
> @thebluepotato 
> 
> 
> 
> > Anyway, you can actually take the "released ZIP" and rename it to XPI and you're good to go. There's a few extra files in there but you should be fine.
> 
> 
> 
> Tried again, does not work. What am I missing?

Can you describe the steps you're taking? What do you mean "rejects", any error message?

SR-- commented:
@thebluepotato 

Thanks!

Steps: Download ZIP for the release. Remove the files that are not in XPI. Rename into XPI. Try to add as a plugin.

Here is the error:

![image](https://github.com/user-attachments/assets/6d4cae40-c43c-494b-81c4-99574dc8f4fb)

And this is the file that I use: https://we.tl/t-UByOVehq5V

Of course the best would be to fix the release...

mjthoraval commented:
@SR-- Which version of Zotero are you using?

mjthoraval commented:
@SR-- The file you have shared contains a folder:
![image](https://github.com/user-attachments/assets/90188e17-db64-4d9e-accf-49449b6a9585)

It should contain the files and folders directly:
![image](https://github.com/user-attachments/assets/bf9882ce-2362-483a-988c-2f7b33269be5)

SR-- commented:
> Which version of Zotero are you using?

7.0.1, and, yes, it works once I remove the folder, silly me. Thanks @mjthoraval
